internal/provider/provider.go

```diff
@@ -19,29 +19,39 @@ func New() *schema.Provider {
 	provider := &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"domain": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("AUTH0_DOMAIN", nil),
```

Comment:
We had to remove the DefaultFunc from the domain because if it set, then Required gets ignored, thus making this attribute Optional, which is not desired.